Sensus.Shared/SensusServiceHelper.cs

```diff
@@ -1298,6 +1310,22 @@ private void RemoveScripts(bool issueNotification, params Script[] scripts)
             {
                 IssuePendingSurveysNotificationAsync(false, false);
             }
+        }
+
+        private bool ScriptIsWithinExclusionWindow(Script script)
+        {
+            bool result = true;
+
+            var runningProtocol = GetRunningProtocolById(script.Runner.Probe.Protocol.Id);
+
+            // check whether the current time is within any of this script's protocol's notification sound exclusion windows.
+            // if we are within a window, don't play a sound.
+            if (runningProtocol != null)
+                foreach (Window window in runningProtocol.NotificationSoundExclusionWindowsList)
+                    if (window.EncompassesCurrentTime())
+                        result = false;
+
+            return result;
```

Comment:
I think this method will be removed once alert-checking code is moved into the notifier.